python/pyspark/ml/tests/test_classification.py

```diff
@@ -283,6 +293,318 @@ def test_logistic_regression(self):
         except OSError:
             pass
 
+    def test_decision_tree_classifier(self):
+        df = (
+            self.spark.createDataFrame(
+                [
+                    (1.0, 1.0, Vectors.dense(0.0, 5.0)),
+                    (0.0, 2.0, Vectors.dense(1.0, 2.0)),
+                    (1.0, 3.0, Vectors.dense(2.0, 1.0)),
+                    (0.0, 4.0, Vectors.dense(3.0, 3.0)),
+                ],
+                ["label", "weight", "features"],
+            )
+            .coalesce(1)
+            .sortWithinPartitions("weight")
```

Comment:
`spark.createDataFrame` output different partitioning between connect and classic,
which slightly affect the training and model coefficients.
Need `coalesce.sortWithinPartitions` to make training deterministic, otherwise, the parity test would fail